charms/mlflow-server/src/charm.py

```diff
@@ -257,6 +234,49 @@ def set_pod_spec(self, event):
         )
         self.model.unit.status = ActiveStatus()
 
+    def _configure_mesh(self, interfaces):
+        if interfaces["ingress"]:
+            interfaces["ingress"].send_data(
+                {
+                    "prefix": "/mlflow",
+                    "rewrite": "/",
+                    "service": self.model.app.name,
+                    "port": self.model.config["mlflow_port"],
+                }
+            )
+
+    def _check_leader(self):
+        if not self.unit.is_leader():
+            # We can't do anything useful when not the leader, so do nothing.
+            raise CheckFailed("Waiting for leadership", WaitingStatus)
+
+    def _get_interfaces(self):
+        try:
+            interfaces = get_interfaces(self)
+        except NoVersionsListed as err:
+            raise CheckFailed(err, WaitingStatus)
+        except NoCompatibleVersions as err:
+            raise CheckFailed(err, BlockedStatus)
+        return interfaces
+
+    def _check_image_details(self):
+        try:
+            image_details = self.image.fetch()
+        except OCIImageResourceError as e:
+            raise CheckFailed(f"{e.status.message}", e.status_type)
+        return image_details
+
+
+class CheckFailed(Exception):
+    """ Raise this exception if one of the checks in main fails. """
```

Comment:
To fix `black` failure
```suggestion
    """ Raise this exception if one of the checks in main fails."""
```

Reply:
That was the issue, thanks @ca-scribner